Game/Source/GDKTestGyms/Private/TestGymsReplicationGraph.cpp

```diff
@@ -0,0 +1,754 @@
+{
+	AActor* CDO = Class->GetDefaultObject<AActor>();
+	if (bSpatialize)
+	{
+		Info.SetCullDistanceSquared(CDO->NetCullDistanceSquared);
+		UE_LOG(LogTestGymsReplicationGraph, Log, TEXT("Setting cull distance for %s to %f (%f)"), *Class->GetName(), Info.GetCullDistanceSquared(), Info.GetCullDistance());
+	}
+
+	Info.ReplicationPeriodFrame = FMath::Max<uint32>( (uint32)FMath::RoundToFloat(ServerMaxTickRate / CDO->NetUpdateFrequency), 1);
+
+	UClass* NativeClass = Class;
+	while(!NativeClass->IsNative() && NativeClass->GetSuperClass() && NativeClass->GetSuperClass() != AActor::StaticClass())
+	{
+		NativeClass = NativeClass->GetSuperClass();
+	}
+
+	UE_LOG(LogTestGymsReplicationGraph, Log, TEXT("Setting replication period for %s (%s) to %d frames (%.2f)"), *Class->GetName(), *NativeClass->GetName(), Info.ReplicationPeriodFrame, CDO->NetUpdateFrequency);
+}
+
+void UTestGymsReplicationGraph::ResetGameWorldState()
+{
+	Super::ResetGameWorldState();
+
+	AlwaysRelevantStreamingLevelActors.Empty();
+
+	for (UNetReplicationGraphConnection* ConnManager : Connections)
+	{
+		for (UReplicationGraphNode* ConnectionNode : ConnManager->GetConnectionGraphNodes())
+		{
+			if (UTestGymsReplicationGraphNode_AlwaysRelevant_ForConnection* AlwaysRelevantConnectionNode = Cast<UTestGymsReplicationGraphNode_AlwaysRelevant_ForConnection>(ConnectionNode))
+			{
+				AlwaysRelevantConnectionNode->ResetGameWorldState();
+			}
+		}
+	}
+
+	for (UNetReplicationGraphConnection* ConnManager : PendingConnections)
+	{
+		for (UReplicationGraphNode* ConnectionNode : ConnManager->GetConnectionGraphNodes())
+		{
+			if (UTestGymsReplicationGraphNode_AlwaysRelevant_ForConnection* AlwaysRelevantConnectionNode = Cast<UTestGymsReplicationGraphNode_AlwaysRelevant_ForConnection>(ConnectionNode))
+			{
+				AlwaysRelevantConnectionNode->ResetGameWorldState();
+			}
+		}
+	}
+}
+
+void UTestGymsReplicationGraph::InitGlobalActorClassSettings()
+{
+	Super::InitGlobalActorClassSettings();
+
+	const bool bUsingSpatial = GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking();
+
+	// ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	// Programatically build the rules.
+	// ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+	
+	auto AddInfo = [&]( UClass* Class, EClassRepNodeMapping Mapping) { ClassRepNodePolicies.Set(Class, Mapping); };
+
+	AddInfo( APlayerState::StaticClass(),							EClassRepNodeMapping::NotRouted);				// Special cased via UTestGymsReplicationGraphNode_PlayerStateFrequencyLimiter
+	AddInfo( AReplicationGraphDebugActor::StaticClass(),			EClassRepNodeMapping::NotRouted);				// Not supported
+	AddInfo( AInfo::StaticClass(),									EClassRepNodeMapping::RelevantAllConnections);	// Non spatialized, relevant to all
```

Comment:
The naming collision between spatialized actors and Spatial as our things is bothering me a lot.